dotnet/src/SemanticKernel.Test/Diagnostics/VerifyTest.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.SemanticKernel.Diagnostics;
+using Xunit;
+
+namespace SemanticKernelTests.Diagnostics;
+
+public class VerifyTests
+{
+    #region Verify.GreaterThan
+    [Fact]
+    public void GreaterThanThrowsWhenValueIsLessThanOrEqualToTarget()
+    {
+        var ex = Assert.Throws<ValidationException>(() => Verify.GreaterThan<int>(-1, 0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.GreaterThan<double>(0.0, 0.0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.GreaterThan<TimeSpan>(TimeSpan.MinValue, TimeSpan.MaxValue));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+    }
+
+    [Fact]
+    public void GreaterThanPassesWhenValueIsGreaterThanTarget()
+    {
+        Verify.GreaterThan<int>(1, 0);
+        Verify.GreaterThan<double>(1.0, 0.0);
+        Verify.GreaterThan<TimeSpan>(TimeSpan.MaxValue, TimeSpan.MinValue);
+    }
+    #endregion
+
+    #region Verify.GreaterThanOrEqualTo
+    [Fact]
+    public void GreaterThanOrEqualToThrowsWhenValueIsLessThanTarget()
+    {
+        var ex = Assert.Throws<ValidationException>(() => Verify.GreaterThanOrEqualTo<int>(-1, 0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.GreaterThan<double>(-0.1, 0.0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.GreaterThan<TimeSpan>(TimeSpan.MinValue, TimeSpan.MaxValue));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+    }
+
+    [Fact]
+    public void GreaterThanOrEqualToPassesWhenValueIsGreaterThanOrEqualToTarget()
+    {
+        Verify.GreaterThanOrEqualTo<int>(1, 0);
+        Verify.GreaterThanOrEqualTo<double>(0.0, 0.0);
+        Verify.GreaterThanOrEqualTo<TimeSpan>(TimeSpan.MaxValue, TimeSpan.MinValue);
+    }
+    #endregion
+
+    #region Verify.LessThan
+    [Fact]
+    public void LessThanThrowsWhenValueIsGreaterThanOrEqualToTarget()
+    {
+        var ex = Assert.Throws<ValidationException>(() => Verify.LessThan<int>(1, 0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.LessThan<double>(0.0, 0.0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.LessThan<TimeSpan>(TimeSpan.MaxValue, TimeSpan.MinValue));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+    }
+
+    public void LessThanPassesWhenValueIsLessThanTarget()
```

Comment:
Can this be removed?

Reply:
This is needed since throwing exceptions on negative cases doesn't guarantee the opposite will pass if the implementation is buggy. A simple case would be the function throws exception regardless of the input.